src/iterators/intoiterator.md

```diff
@@ -0,0 +1,86 @@
+---
+minutes: 5
+existing course material:
+- exercises/day-2/iterators-and-ownership.md
+---
+
+# `IntoIterator`
+
+The `Iterator` trait tells you how to _iterate_ once you have created an
+iterator. The related trait
+[`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html)
+defines how to create an iterator for a type. It is used automatically by the
+`for` loop.
```

Comment:
This applies to many of the new pages: most of the old pages (at least those I wrote) followed a scheme of

1. A single line of text to introduce the slide
2. A small code block
3. Perhaps 2-3 lines of extra text before the notes start

You see that in [6.3. References](https://google.github.io/comprehensive-rust/basic-syntax/references.html) and the next few slides.

Wny do they look like this? Becuase that's who I've been writing all my slides for years :smile: 

I don't know if you're aware of the "aspect ratio helper"? Uncomment it in `book.toml` and it'll draw a big rectangle on top of the pages. It shows that code blocks with more than 15 lines are too large — as soon as you execute them and get 1-2 lines of results, then people lose the overview:

![image](https://github.com/google/comprehensive-rust/assets/89623/142c2fc8-5111-4571-a145-c45ab66798e4)